src/main.c

```diff
@@ -2408,15 +2409,58 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/9)
+/// This function returns the movement speed based on whether the character should teleport.
+///
+///\param teleport : whether the character should teleport
+///\param teleportHeldLastFrame : whether the teleport button was previously held
+///\param groundedMovementSpeed : default movement speed
+
+int getGroundedMovementSpeed(bool teleport, bool *teleportHeldLastFrame, int groundedMovementSpeed)
+{
+    if (teleport)
+    {
+        if (*teleportHeldLastFrame == false){
+            *teleportHeldLastFrame = true;
+            return 600;
+        }
+        *teleportHeldLastFrame = true;
+    }
+    else
+    {
+        *teleportHeldLastFrame = false;
+    }
+
+    return groundedMovementSpeed;
+}
+
+void handle_movement(double dt, bool *teleportHeldLastFrame)
+{
     static float dy = 0;
     State *s = &g->players->state;
     int sz = 0;
     int sx = 0;
+    char shouldTeleport = 0;
```

Comment:
What was the reason for using a char for this variable? it seems at first glance that it could be a bool.

Reply:
Other than That I cannot see anything else wrong. Should be good to merge